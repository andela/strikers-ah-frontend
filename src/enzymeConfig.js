import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
