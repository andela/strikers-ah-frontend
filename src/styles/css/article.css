@import url("https://fonts.googleapis.com/css?family=Raleway&display=swap");
.spiner {
  position: fixed;
  z-index: 1031;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: #43c0f1; }

.comments-container {
  padding: 1%;
  margin-top: 1%;
  border-radius: 1%; }
  .comments-container .comment-count {
    font-weight: bold;
    margin-left: 4%; }

html {
  height: 100%;
  margin: 0; }

body {
  font-family: "Arial", "Helvetica", "sans-serif";
  font-size: 15px;
  margin: 0;
  padding: 0;
  /* background-color: darkgray; */
  min-height: 100%; }

.container {
  width: 80%;
  margin: auto;
  overflow: hidden; }
  @media (max-width: 768px) {
    .container {
      width: 100%; } }

header {
  background-color: #ffffff;
  padding-top: 10px;
  min-height: 70px;
  border-bottom: #e5e5e5 3px solid; }

.brand {
  margin-top: 10px;
  float: left; }

header .icons ul {
  list-style-type: none;
  margin: 0;
  padding: 0; }

header .icons ul li i {
  color: grey; }

header .icons {
  margin-top: 10px;
  display: flex;
  flex-direction: row;
  justify-content: flex-end; }
  @media (max-width: 768px) {
    header .icons {
      display: none; } }

.brand .brandName {
  text-decoration: none; }

.brand .brandName {
  font-size: 25px;
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #00aef9; }

.content {
  background-color: #e5e5e5;
  align-content: center;
  margin-right: 100px;
  margin-left: 100px; }
  @media (max-width: 768px) {
    .content {
      margin: auto;
      margin-right: 5%;
      margin-left: 5%; } }

.content .editor-tools .container .bar-tool #align-bartool {
  text-align: center; }

.content .editor-tools .container .bar-tool {
  margin-top: 10px;
  margin-bottom: 10px;
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: 0px 4px 4px #c4c4c4;
  padding: 10px;
  display: inline-block;
  /* position: absolute; */ }
  @media (max-width: 768px) {
    .content .editor-tools .container .bar-tool {
      margin: auto;
      padding: 5px; } }

.content .editor-tools .container .bar-tool {
  margin-bottom: 100px;
  margin-left: 10%;
  margin-right: 10%; }

input[type='color'] {
  border: none; }

.content .editor-tools .container .bar-tool #border {
  border: 1px solid #c4c4c4;
  transform: rotate(90deg); }

.content .editor-tools .container .bar-tool > button {
  margin-right: 5px;
  margin-left: 5px;
  border: no; }

.content .editor-tools .container .bar-tool > button:hover {
  background-color: #dfe6e9;
  transition: all 0.3s linear 0s;
  cursor: pointer; }

.content .editor-tools .container .bar-tool #highlight {
  cursor: pointer; }

.content .editor-tools .container .bar-tool #highlight {
  margin-right: 5px;
  margin-left: 5px; }

.content .text-area {
  width: 100%; }

[contentEditable='true']:empty:not(:focus):before {
  content: attr(data-text);
  color: gray; }

.content .text-area .form #title {
  background-color: #ffffff;
  margin-bottom: 5px;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: "Arial", "Helvetica", "sans-serif";
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%; }

.content .text-area .form #body {
  margin-bottom: 10px;
  background-color: #ffffff;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: "Arial", "Helvetica", "sans-serif";
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
  min-height: 800px; }

.content .text-area .form #publish {
  cursor: pointer;
  border: none;
  background: #808e9b;
  font-size: 16px;
  padding: 15px 32px;
  display: inline-block;
  text-align: center;
  color: #ffffff;
  border-radius: 5px;
  float: right;
  margin-bottom: 20px; }

nav.icons div {
  margin-right: 15px; }

nav.icons i {
  margin-top: 5px; }

nav.icons div.search input[type='text'] {
  margin-right: 10px;
  border-radius: 6px;
  padding: 4px 12px;
  border-color: #eee;
  border-width: 1px; }

nav.icons div.search #search-button {
  margin-right: 10px; }

.card {
  float: left;
  margin-top: 10px;
  padding: 10px;
  width: 25%;
  height: 400px;
  background-color: #ffffff;
  border-color: #00aef9;
  border-width: 1px;
  box-shadow: #808e9b;
  margin-right: 10px; }

.card:hover {
  cursor: pointer;
  text-decoration: none; }

.card > img {
  width: 100%;
  height: 50%; }

.card .contentDescription {
  width: 100%;
  margin-top: 10px;
  word-wrap: break-word; }

.card .contentDescription .articleCardHeader {
  width: 100%;
  text-align: center;
  word-wrap: break-word; }

.card .contentDescription .articleCardHeader p {
  font-weight: bold;
  font-size: 20px;
  text-transform: capitalize; }

div.alert {
  width: 30%;
  height: 30px;
  display: inline-block;
  background-color: #bb321f;
  color: #ffffff;
  margin-left: 30%;
  margin-right: 30%;
  border-radius: 5px;
  margin-top: 10px;
  padding: 30px;
  font-size: 20px; }

.contents {
  background-color: #ffffff;
  margin-top: 0;
  min-height: 800px; }

.contents .contentTitle {
  width: 100%;
  text-align: center;
  padding: 10px; }

.contents .contentBody #articleBody {
  clear: both;
  float: left;
  margin-top: 15px;
  padding-left: 10%;
  padding-right: 10%; }

.contents .contentBody p {
  padding-left: 50px;
  padding-right: 50px;
  font-size: 20px;
  word-wrap: break-word;
  line-height: 1.5em; }

.contents .contentBody p img {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  display: block;
  height: 500px; }

.article-icons .icon {
  margin-right: 10px;
  display: inline-block; }

.contentDescription a {
  text-decoration: none; }

.contentDescription a:link {
  color: #2f3640; }

.contentDescription a:visited {
  color: #2f3640; }

.card .contentDescription .buttons {
  margin-bottom: 5px;
  position: absolute;
  margin-top: 10px; }

.card .contentDescription .buttons .redButton {
  background: #bb321f;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px; }

.card .contentDescription .buttons .lightBlueButton {
  background: #74b9ff;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px; }

.card .articleDescription {
  height: 65px;
  margin-top: 5px; }

.container .allArticlecontents {
  background-color: #ffffff;
  margin-top: 0;
  height: 100%; }

.socialMedia {
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer; }
  .socialMedia .facebookButton {
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
    .socialMedia .facebookButton.facebook {
      color: #3b5998;
      border-color: #3b5998; }
      .socialMedia .facebookButton.facebook:hover {
        background-color: #3b5998;
        border-color: #3b5998;
        color: #fff; }
    .socialMedia .facebookButton.twitter {
      color: #1da1f2;
      border-color: #1da1f2; }
      .socialMedia .facebookButton.twitter:hover {
        background-color: #1da1f2;
        border-color: #1da1f2;
        color: #fff; }
    .socialMedia .facebookButton.googleplus {
      color: #dd4b39;
      border-color: #dd4b39; }
      .socialMedia .facebookButton.googleplus:hover {
        background-color: #dd4b39;
        border-color: #dd4b39;
        color: #fff; }
    .socialMedia .facebookButton.linkedin {
      color: #0077b5;
      border-color: #0077b5; }
      .socialMedia .facebookButton.linkedin:hover {
        background-color: #0077b5;
        border-color: #0077b5;
        color: #fff; }
    .socialMedia .facebookButton.whatsapp {
      color: #128c7e;
      border-color: #128c7e; }
      .socialMedia .facebookButton.whatsapp:hover {
        background-color: #128c7e;
        border-color: #128c7e;
        color: #fff; }
  .socialMedia .twitterButton {
    color: #1da1f2;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .twitterButton:hover {
    background-color: #1da1f2;
    color: #fff; }
  .socialMedia .linkedinButton {
    color: #0077b5;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .linkedinButton:hover {
    background-color: #0077b5;
    color: #fff; }
  .socialMedia .whatsappButton {
    color: #128c7e;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .whatsappButton:hover {
    color: #fff;
    background-color: #128c7e; }
  .socialMedia .emailButton {
    color: #dd4b39;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .emailButton:hover {
    color: #fff;
    background-color: #dd4b39; }

.edit-history-section {
  margin-top: 10px; }
  .edit-history-section .hideButton {
    background: #fff;
    padding: 3px;
    color: #000;
    border-radius: 3px;
    cursor: pointer;
    margin-bottom: 3px;
    font-size: 20px; }

.history-section {
  padding: 3px 15px 5px 0px;
  border-bottom: 1px solid #c5c5c5; }
  .history-section time {
    color: #9c9c9c;
    font-size: 12px;
    font-weight: bold; }

.emailButton:hover {
  color: #fff;
  background-color: #dd4b39; }

.likeButton {
  border: none;
  color: #9c9c9c;
  cursor: pointer;
  width: 15px;
  height: 15px; }

.likeButton:hover {
  color: #43c0f1; }

.buttonSpan {
  margin-right: 5px; }

.topImage input[type='file'] {
  cursor: inherit;
  display: block;
  min-height: 100%;
  min-width: 100%;
  opacity: 0;
  position: absolute;
  right: 0;
  text-align: right;
  top: 0;
  float: right; }

.topImage .file-upload {
  overflow: hidden;
  position: absolute;
  width: 100px;
  height: 100px; }

.topImage .file-upload .imageIcone {
  min-height: 100%;
  min-width: 100%;
  margin-left: 20%;
  color: #3b5998; }

.tags {
  padding: 10px;
  margin-left: 20px;
  background-color: #3b5998;
  color: #fff;
  border-radius: 5px;
  display: inline-block;
  cursor: pointer; }

#tags > input {
  border-radius: 5px;
  border: none;
  width: 178px;
  height: 36px;
  margin-bottom: 5px;
  padding: 10px;
  font-size: 15px;
  margin-right: 5px; }

#tags {
  width: 75%;
  float: left; }

#addTag {
  width: 10%;
  float: left; }

.tagList {
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
  background-color: #4b4b4b;
  color: #fff;
  display: inline-block;
  padding: 5px;
  cursor: pointer; }
