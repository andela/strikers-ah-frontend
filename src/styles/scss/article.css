@import url('https://fonts.googleapis.com/css?family=Raleway&display=swap');
.comments-container {
  padding: 1%;
  margin-top: 1%;
  border-radius: 1%;
}

.comments-container .comment-count {
  font-weight: bold;
  margin-left: 4%;
}

html {
  height: 100%;
  margin: 0;
}

body {
  font-family: 'Arial', 'Helvetica', 'sans-serif';
  font-size: 15px;
  margin: 0;
  padding: 0;
  /* background-color: darkgray; */
  min-height: 100%;
}

.container {
  width: 80%;
  margin: auto;
  overflow: hidden;
}

@media (max-width: 768px) {
  .container {
    width: 100%;
  }
}

header {
  background-color: #ffffff;
  padding-top: 10px;
  min-height: 70px;
  border-bottom: #e5e5e5 3px solid;
}

.brand {
  margin-top: 10px;
  float: left;
}

header .icons ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

header .icons ul li i {
  color: grey;
}

header .icons {
  margin-top: 10px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

@media (max-width: 768px) {
  header .icons {
    display: none;
  }
}

.brand .brandName {
  text-decoration: none;
}

.brand .brandName {
  font-size: 25px;
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #00aef9;
}

.content {
  background-color: #e5e5e5;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-right: 100px;
  margin-left: 100px;
}

@media (max-width: 768px) {
  .content {
    margin: auto;
    margin-right: 5%;
    margin-left: 5%;
  }
}

.content .editor-tools .container .bar-tool #align-bartool {
  text-align: center;
}

.content .editor-tools .container .bar-tool {
  margin-top: 10px;
  margin-left: 30%;
  margin-right: 30%;
  margin-bottom: 10px;
  background-color: #ffffff;
  border-radius: 8px;
  -webkit-box-shadow: 0px 4px 4px #c4c4c4;
  box-shadow: 0px 4px 4px #c4c4c4;
  padding: 10px;
  display: inline-block;
  /* position: absolute; */
}

@media (max-width: 768px) {
  .content .editor-tools .container .bar-tool {
    margin: auto;
    padding: 5px;
  }
}

input[type='color'] {
  border: none;
}

.content .editor-tools .container .bar-tool #border {
  border: 1px solid #c4c4c4;
  -webkit-transform: rotate(90deg);
  transform: rotate(90deg);
}

.content .editor-tools .container .bar-tool > button {
  margin-right: 5px;
  margin-left: 5px;
  border: no;
}

.content .editor-tools .container .bar-tool > button:hover {
  background-color: #dfe6e9;
  -webkit-transition: all 0.3s linear 0s;
  transition: all 0.3s linear 0s;
  cursor: pointer;
}

.content .editor-tools .container .bar-tool #highlight {
  cursor: pointer;
}

.content .editor-tools .container .bar-tool #highlight {
  margin-right: 5px;
  margin-left: 5px;
}

.content .text-area {
  width: 100%;
}

[contentEditable='true']:empty:not(:focus):before {
  content: attr(data-text);
  color: gray;
}

.content .text-area .form #title {
  background-color: #ffffff;
  margin-bottom: 5px;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: 'Arial', 'Helvetica', 'sans-serif';
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
}

.content .text-area .form #body {
  margin-bottom: 10px;
  background-color: #ffffff;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: 'Arial', 'Helvetica', 'sans-serif';
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
  min-height: 800px;
}

.content .text-area .form #publish {
  border: none;
  background: #808e9b;
  font-size: 16px;
  padding: 15px 32px;
  display: inline-block;
  text-align: center;
  color: #ffffff;
  border-radius: 5px;
  float: right;
  margin-bottom: 20px;
}

nav.icons div {
  margin-right: 15px;
}

nav.icons i {
  margin-top: 5px;
}

nav.icons div.search input[type='text'] {
  margin-right: 10px;
  border-radius: 6px;
  padding: 4px 12px;
  border-color: #eee;
  border-width: 1px;
}

nav.icons div.search #search-button {
  margin-right: 10px;
}

.card {
  float: left;
  margin-top: 10px;
  padding: 10px;
  width: 25%;
  height: 400px;
  background-color: #ffffff;
  border-color: #00aef9;
  border-width: 1px;
  -webkit-box-shadow: #808e9b;
  box-shadow: #808e9b;
  margin-right: 10px;
}

.card:hover {
  cursor: pointer;
  text-decoration: none;
}

.card > img {
  width: 100%;
  height: 50%;
}

.card .contentDescription {
  width: 100%;
  margin-top: 10px;
  word-wrap: break-word;
}

.card .contentDescription .articleCardHeader {
  width: 100%;
  text-align: center;
  word-wrap: break-word;
}

.card .contentDescription .articleCardHeader p {
  font-weight: bold;
  font-size: 20px;
  text-transform: capitalize;
}

div.alert {
  width: 30%;
  height: 30px;
  display: inline-block;
  background-color: #bb321f;
  color: #ffffff;
  margin-left: 30%;
  margin-right: 30%;
  border-radius: 5px;
  margin-top: 10px;
  padding: 30px;
  font-size: 20px;
}

.contents {
  background-color: #ffffff;
  margin-top: 0;
  min-height: 800px;
}

.contents .contentTitle {
  width: 100%;
  text-align: center;
  padding: 10px;
}

.contents .contentBody p {
  padding-left: 50px;
  padding-right: 50px;
  font-size: 20px;
  word-wrap: break-word;
}

.contents .contentBody p img {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  display: block;
  height: 500px;
}

.article-icons .icon {
  margin-right: 10px;
  display: inline-block;
}

.contentDescription a {
  text-decoration: none;
}

.contentDescription a:link {
  color: #2f3640;
}

.contentDescription a:visited {
  color: #2f3640;
}

.card .contentDescription .buttons {
  margin-bottom: 5px;
  position: absolute;
  margin-top: 10px;
}

.card .contentDescription .buttons .redButton {
  background: #bb321f;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px;
}

.card .contentDescription .buttons .lightBlueButton {
  background: #74b9ff;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px;
}

.card .articleDescription {
  height: 65px;
  margin-top: 5px;
}

.container .allArticlecontents {
  background-color: #ffffff;
  margin-top: 0;
  height: 100%;
}

.socialMedia {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.socialMedia .facebookButton {
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .facebookButton.facebook {
  color: #3b5998;
  border-color: #3b5998;
}

.socialMedia .facebookButton.facebook:hover {
  background-color: #3b5998;
  border-color: #3b5998;
  color: #fff;
}

.socialMedia .facebookButton.twitter {
  color: #1da1f2;
  border-color: #1da1f2;
}

.socialMedia .facebookButton.twitter:hover {
  background-color: #1da1f2;
  border-color: #1da1f2;
  color: #fff;
}

.socialMedia .facebookButton.googleplus {
  color: #dd4b39;
  border-color: #dd4b39;
}

.socialMedia .facebookButton.googleplus:hover {
  background-color: #dd4b39;
  border-color: #dd4b39;
  color: #fff;
}

.socialMedia .facebookButton.linkedin {
  color: #0077b5;
  border-color: #0077b5;
}

.socialMedia .facebookButton.linkedin:hover {
  background-color: #0077b5;
  border-color: #0077b5;
  color: #fff;
}

.socialMedia .facebookButton.whatsapp {
  color: #128c7e;
  border-color: #128c7e;
}

.socialMedia .facebookButton.whatsapp:hover {
  background-color: #128c7e;
  border-color: #128c7e;
  color: #fff;
}

.socialMedia .twitterButton {
  color: #1da1f2;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .twitterButton:hover {
  background-color: #1da1f2;
  color: #fff;
}

.socialMedia .linkedinButton {
  color: #0077b5;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .linkedinButton:hover {
  background-color: #0077b5;
  color: #fff;
}

.socialMedia .whatsappButton {
  color: #128c7e;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .whatsappButton:hover {
  color: #fff;
  background-color: #128c7e;
}

.socialMedia .emailButton {
  color: #dd4b39;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .emailButton:hover {
  color: #fff;
  background-color: #dd4b39;
}

.edit-history-section {
  margin-top: 10px;
}

.edit-history-section .hideButton {
  background: #fff;
  padding: 3px;
  color: #000;
  border-radius: 3px;
  cursor: pointer;
  margin-bottom: 3px;
  font-size: 20px;
}

.history-section {
  padding: 3px 15px 5px 0px;
  border-bottom: 1px solid #c5c5c5;
}

.history-section time {
  color: #9c9c9c;
  font-size: 12px;
  font-weight: bold;
}

.emailButton:hover {
  color: #fff;
  background-color: #dd4b39;
}

.likeButton {
  border: none;
  color: #9c9c9c;
  cursor: pointer;
  width: 15px;
  height: 15px;
}

.likeButton:hover {
  color: #43c0f1;
}

.buttonSpan {
  margin-right: 5px;
}
/*# sourceMappingURL=article.css.map */
