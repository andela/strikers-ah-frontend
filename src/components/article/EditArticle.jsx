/* eslint-disable no-plusplus */
/* eslint-disable no-lone-blocks */
/* eslint-disable no-unused-expressions */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable no-unused-vars */
/* eslint-disable prefer-destructuring */
/* eslint-disable react/no-access-state-in-setstate */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable react/no-unused-state */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable dot-notation */
/* eslint-disable no-return-assign */
import React, { Component, Fragment } from 'react';
import { Redirect } from 'react-router-dom';
import sweeetAlert from 'sweetalert';
import reactHtmlparser from 'html-react-parser';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import ReactQuill, { Quill } from 'react-quill';
import { faImage } from '@fortawesome/free-solid-svg-icons';
import 'react-quill/dist/quill.snow.css';
import 'quill/dist/quill.snow.css';
import 'font-awesome/css/font-awesome.min.css';
import {
  getOneArticle,
  updateArticle,
} from '../../redux/actions/articleAction';
import alert from '../../redux/actions/alert';
import Header from './Header';
import UpdateMessageDisplay from './UpdateMessageDisplay';

import '../../styles/css/article.css';
import Alerts from './Alert';

/**
 * @description styling the bold button
 */

export const EditorBar = () => {
  return (
    <div className="content">
      <Alerts />
      <section className="editor-tools">
        <div className="container">
          <div id="align-bartool">
            <div className="bar-tool">
              <div id="toolbar">
                <button type="button" className="ql-bold" />
                <button type="button" className="ql-italic" />
                <button type="button" className="ql-underline" />
                <button type="button" className="ql-strike" />
                <button type="button" className="ql-blockquote" />
                <button type="button" className="ql-code-block" />
                <select className="ql-font">
                  <option value="arial" selected>
                    Arial
                  </option>
                  <option value="comic-sans">Comic Sans</option>
                  <option value="courier-new">Courier New</option>
                  <option value="georgia">Georgia</option>
                  <option value="helvetica">Helvetica</option>
                  <option value="lucida">Lucida</option>
                </select>
                <select className="ql-size">
                  <option value="extra-small">Size 1</option>
                  <option value="small">Size 2</option>
                  <option value="medium" selected>
                    Size 3
                  </option>
                  <option value="large">Size 4</option>
                </select>
                <button type="button" className="ql-link" />
                <select className="ql-align" />
                <select className="ql-color" />
                <select className="ql-background" />
                <button type="button" className="ql-clean" />
                <button type="button" className="ql-image" />
                <button type="button" className="ql-video" />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  );
};
const Size = Quill.import('formats/size');
Size.whitelist = ['extra-small', 'small', 'medium', 'large'];
Quill.register(Size, true);

// Add fonts to whitelist and register them
const Font = Quill.import('formats/font');
Font.whitelist = [
  'arial',
  'comic-sans',
  'courier-new',
  'georgia',
  'helvetica',
  'lucida',
];
Quill.register(Font, true);

/*
 * Editor component with custom toolbar and content containers
 */
/**
 * @author Innocent Nkunzi
 * @param {*} slug
 * @description editor
 * @returns {*} class
 *
 */
export class EditArticle extends React.Component {
  static modules = {
    toolbar: {
      container: '#toolbar',
    },
  };

  static formats = [
    'header',
    'font',
    'size',
    'bold',
    'italic',
    'underline',
    'strike',
    'blockquote',
    'list',
    'bullet',
    'indent',
    'link',
    'image',
    'color',
    'code-block',
  ];

  /**
   *
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      body: '',
      image: null,
      taglist: [],
      tagInputs: [{ name: 'input0', value: '' }],
      slug: '',
    };
    this.BodyContent = React.createRef();
  }

  /**
   * @description it gets one article first
   * @returns {*} --
   */
  componentWillMount() {
    const {
      match: { params },
    } = this.props;
    this.props.getOneArticle(params.slug);
  }

  /**
   * @description it gets one article first
   * @returns {*} --
   * @param {*} nextProps
   */
  componentWillReceiveProps(nextProps) {
    const { article } = nextProps;
    this.setState({
      title: article.title,
      body: article.body,
      slug: article.slug,
    });
  }

  onSubmit = slug => {
    const { title, body, image, taglist } = this.state;
    const data = {
      title,
      body,
      taglist,
      image,
    };

    this.props.updateArticle(data, slug);
  };

  handleChange = ({ target }) => {
    const { value, name } = target;
    this.setState({ [name]: value });
  };

  handleImage = ({ target }) => {
    this.setState({ image: target.files[0] });
  };

  /**
   * @param { * } body --
   * @returns { * }--
   */
  handleQuillChange = body => {
    this.setState({ body });
  };

  tagInputChange = ({ target }) => {
    const { name, value } = target;
    const tagInput = this.state.tagInputs.find(input => input.name === name);
    const index = this.state.tagInputs.indexOf(tagInput);
    const { tagInputs } = this.state;
    tagInputs[index].value = value;
    this.setState({ tagInputs });
  };

  makeInput = () => {
    const { tagInputs } = this.state;
    const newInput = { name: `input${tagInputs.length}`, value: '' };
    tagInputs.push(newInput);
    const inputWithValues = this.state.tagInputs.filter(
      input => input.value.length > 0,
    );
    const taglist = inputWithValues.map(input => input.value);
    this.setState({ tagInputs, taglist });
  };

  /**
   * @author Innocent Nkunzi
   * @returns {*} editor
   */
  render() {
    let id = 1;
    const { article } = this.props;
    const { title, body } = this.state;
    {
      article === undefined || (article === null && <p>{article.error}</p>);
    }

    const { tagInputs } = this.state;
    return (
      <div className="content">
        <EditorBar />
        <UpdateMessageDisplay />
        <section className="text-area">
          <div className="topImage">
            <label htmlFor="file-upload" className="file-upload">
              <FontAwesomeIcon icon={faImage} className="imageIcone" />
              <input
                type="file"
                id="sigleImage"
                onChange={this.handleImage}
                name="image"
              />
            </label>
          </div>
          <div className="container">
            <form className="form">
              <div>
                <input
                  type="text"
                  name="title"
                  value={this.state.title || ''}
                  onChange={this.handleChange}
                  placeholder="Title goes here"
                  id="title"
                  maxLength="80"
                />
              </div>
              <div>
                <ReactQuill
                  id="body"
                  value={this.state.body || ''}
                  onChange={this.handleQuillChange}
                  placeholder={this.props.placeholder}
                  modules={EditArticle.modules}
                  formats={EditArticle.formats}
                  name="body"
                />
              </div>
              <button
                type="button"
                onClick={() => this.onSubmit(article.slug)}
                id="publish"
              >
                Publish
              </button>
            </form>
            <React.Fragment>
              <div id="addTag">
                <button
                  type="button"
                  id="makeInput"
                  onClick={this.makeInput}
                  className="tags"
                >
                  AddTag
                </button>
              </div>
              <div id="tags">
                {tagInputs.map(input => (
                  <input
                    type="text"
                    id="tagInputchange"
                    name={input.name}
                    placeholder="#"
                    onChange={this.tagInputChange}
                    value={input.value}
                    key={id++}
                  />
                ))}
              </div>
            </React.Fragment>
          </div>
        </section>
      </div>
    );
  }
}

const mapStateToprops = state => ({
  article: { ...state.Article.article },
});

export default connect(
  mapStateToprops,
  { alert, getOneArticle, updateArticle },
)(EditArticle);
