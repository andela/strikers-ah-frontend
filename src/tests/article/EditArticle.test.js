import React from 'react';
import { shallow } from 'enzyme';
import '../../enzymeConfig';

import { EditArticle, EditorBar } from '../../components/article/EditArticle';

const event = {
  target: { value: 'title', name: 'input0' },
  tagInputs: [{ name: 'input0', value: 'fasdf' }],
};

const imageEvent = {
  target: {
    files: ['just an array'],
  },
  image: [],
};

const props = {
  title: 'this is the title',
  body: 'the body',
  slug: 'sorting-array-in-a-ascending-order-32488f8',
  taglist: [],
  match: {
    params: jest.fn(),
  },
  getOneArticle: jest.fn(),
  onSubmit: jest.fn(),
  article: {
    body: 'fhksjfhkahkfjah',
  },
  alert: jest.fn(),
  target: { name: 'special', value: 'party' },
  updateArticle: jest.fn(),
  data: {
    title: '',
    body: '',
  },
};
const wrapper = shallow(<EditArticle {...props} />);

describe('Edit article test', () => {
  test('should render the `EditArticle` component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should render the `EditArticle` component', () => {
    const Wrapper = shallow(<EditorBar />);
    expect(Wrapper).toMatchSnapshot();
  });
  test('should test the `onSubmit` button', () => {
    const spy = jest.spyOn(wrapper.instance(), 'onSubmit');
    wrapper.instance().forceUpdate();
    const formInput = wrapper.find('#publish');
    formInput.simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
  test('should test the `handleChange` method', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleChange');
    wrapper.instance().forceUpdate();
    const handlechange = wrapper.find('#title');
    handlechange.simulate('change', event);
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
  test('should test the `handleImage` method', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleImage');
    wrapper.instance().forceUpdate();
    const handlechange = wrapper.find('#sigleImage');
    handlechange.simulate('change', imageEvent);
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
  test('should test the `handleQuillChange` method', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleQuillChange');
    wrapper.instance().forceUpdate();
    const handlechange = wrapper.find('#body');
    handlechange.simulate('change', event);
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
  test('should test the `makeInput` method', () => {
    const spy = jest.spyOn(wrapper.instance(), 'makeInput');
    wrapper.instance().forceUpdate();
    const handlechange = wrapper.find('#makeInput');
    handlechange.simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
  test('should test the `tagInputChange` method', () => {
    const spy = jest.spyOn(wrapper.instance(), 'tagInputChange');
    wrapper.instance().forceUpdate();
    wrapper
      .find('#tagInputchange')
      .at(0)
      .simulate('change', event);
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
});
