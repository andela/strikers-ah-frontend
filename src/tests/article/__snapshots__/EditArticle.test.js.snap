// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit article test should render the \`EditArticle\` component 1`] = `ShallowWrapper {}`;

exports[`Edit article test should render the \`EditArticle\` component 2`] = `ShallowWrapper {}`;
